libraries/lib-files/FileNames.cpp

```diff
@@ -224,13 +233,69 @@ wxString FileNames::LowerCaseAppNameInPath( const wxString & dirIn){
    return dir;
 }
 
-FilePath FileNames::DataDir()
+enum class DirTarget
 {
+    Cache,
+    Config,
+    Data,
+    State
+};
+
+FilePath &gTargetDir(DirTarget target)
```

Comment:
Can be done as
```
struct XDGDirConfig final
{
   wxString envName;
   wxString defaultDir;
};

XDGDirConfig xdgConfig[] = {
  { L"XDG_CACHE_HOME", L"/.cache" },
...
};

static_assert(sizeof(xdgConfig) / sizeof(xdgConfig[0] == size_t(DirTarget::Count)

wxString GetXDGDir(DirTarget target)
{
   const auto& dirConfig = xdgConfig[size_t(target)];
 
   wxString dir;
   // see if the XDG_*_HOME env var is defined. if it is, use its value.
   // if it isn't, use the default XDG-specified value.
   if ( !wxGetEnv(dirConfig.envName, &dir) || dir.empty() )
        dir = wxFileName::GetHomeDir() +  dirConfig.defaultDir;

  return dir;
}
```

This way we force the compiler to do all the checks necessary. Given that the enum is private, there is not much need left for any runtime checks.

Reply:
How about
```c++
using XDGDirConfig = std::tuple<...>;
```
Also I think the `xdgConfig` array should be `const`.

Reply:
std::pair would do too, but I prefer descriptive names :-)